test/helpers/kubectl.go

```diff
@@ -2825,6 +2819,19 @@ func (kub *Kubectl) reportMapHost(ctx context.Context, path string, reportCmds m
 	}
 }
 
+// HelmTemplate renders given helm template. TODO: use go helm library for that
+func (kub *Kubectl) HelmTemplate(chartDir, namespace, filename string, options map[string]string) *CmdRes {
+	optionsString := ""
+
+	for k, v := range options {
+		optionsString += fmt.Sprintf(" --set %s=%s ", k, v)
+	}
+
+	return kub.ExecMiddle("helm template " +
+		chartDir + " " +
+		fmt.Sprintf("--namespace=%s %s > %s", namespace, optionsString, filename))
```

Comment:
Can we avoid redirecting the output to a file? we can do `helm.... | kubectl apply --dry-run -f - ` to get the output of helm template into our logs and then do `helm.... | kubectl apply -f - ` to install it in kubernetes.

Reply:
We could. This is how the older invocations worked, though. Some check that the helm generation worked by checking the file (https://github.com/cilium/cilium/pull/9853/files#diff-2304891712e5d0140828499fc3a19043L1329-L1333 and https://github.com/cilium/cilium/pull/9853/files#diff-912512efc6ed306b1ab20cd37f6db86fL323-L331).
How much do you prefer not using a file? Switching to that might end up a little more complicated since the apply of the yaml happens separately. Were you thinking that we would apply in this function always?

Reply:
@raybejjani I'm just afraid that we are relying on files to get the generated result of helm. If 2 threads use this function with different options they will overwrite the same file.

Reply:
@aanm Right now, ginkgo is single-threaded but, yes, this could come up at some point if ginkgo ever became more parallel (as I recall they were very against the idea). The different parallel runs are in different directories, and these files should be in those directories, not in the global namespace.
Looking at the callsites for this, I could try to randomise the filenames. It means generating the random name and passing it down since it needs to be known further up in some cases. Would that make you more comfortable?

Reply:
@raybejjani yes!